expertise/service/expertise.py

```diff
@@ -368,13 +371,14 @@ def _prepare_config(self, request, job_id=None, client_v1=None, client=None) ->
         self.logger.info(f"Config validation passed - {config.to_json()}")
 
         # Create directory and config file
-        if not os.path.isdir(config.dataset['directory']):
-            os.makedirs(config.dataset['directory'])
-        with open(os.path.join(config.job_dir, 'config.json'), 'w+') as f:
-            json.dump(config.to_json(), f, ensure_ascii=False, indent=4)
-        if not self.containerized:
-            self.logger.info(f"Saving processed config to {os.path.join(config.job_dir, 'config.json')}")
-            self.redis.save_job(config)
+        if save:
+            if not os.path.isdir(config.dataset['directory']):
+                os.makedirs(config.dataset['directory'])
+            with open(os.path.join(config.job_dir, 'config.json'), 'w+') as f:
+                json.dump(config.to_json(), f, ensure_ascii=False, indent=4)
+            if not self.containerized:
+                self.logger.info(f"Saving processed config to {os.path.join(config.job_dir, 'config.json')}")
+                self.redis.save_job(config)
```

Comment:
why don't we save the job by id in redis so we can override it if has the same id?

Reply:
They do get saved by `config.job_id` but if `JobConfig.from_request` gets a None `job_id` or doesn't get passed a `job_id`, it creates generates a new one

Reply:
why the job_id is None?

Reply:
The job_id is None when we need to make a new configuration for a new job, do you mean for here when I need to parse the request again so that I can get the number of submissions?
https://github.com/openreview/openreview-expertise/blob/547a7dfc8892751ba99e7378485c7dda66934d4a/expertise/service/expertise.py#L833-L838
Yeah actually I think I can just pass the job_id here